Can oci-archive use a network location?

Hello 👋 

Is it possible to make `image` work with network locations?

Example is: 
```hcl
    task "my-http" {
      driver = "podman"
      config {
        image = "oci-archive:http://10.211.55.9:9999/uploads/myhttp.tar"
      }
    }
```

Where is `myhttp.tar` is an archive that I serve via HTTP.

I created it via `podman save --format oci-archive -o /tmp/uploads/myhttp.tar myhttp:latest`.

Note, that podman itself supports getting archive be it filesystem or a network location. Consider this:

```
podman load -i http://10.211.55.9:9999/uploads/myhttp.tar
Getting image source signatures
Copying blob 99cf8ab899fa skipped: already exists  
Copying blob 41cbe294e54d skipped: already exists  
Copying blob 870027e70bc9 skipped: already exists  
Copying blob fc8a6b71fc5e skipped: already exists  
Copying config 58e9d534e8 done  
Writing manifest to image destination
Storing signatures
Loaded image: myhttp:latest
`

Just a heads up that I took a look at this and wasn't able to make it work by passing that `oci-archive` transport like this. The error you get is like the following:

```
2025-09-17T16:26:43.070-0400 [ERROR] client.alloc_runner.task_runner: running driver failed: alloc_id=819866eb-1f6d-433d-8d35-3ca5f3ba8d0a task=http error="rpc error: code = Unknown desc = failed to create image: oci-archive:http://192.168.1.194:9999/busybox.tar: invalid image reference oci-archive:http://192.168.1.194:9999/busybox.tar: Invalid image //192.168.1.194:9999/busybox.tar"
```

That's bubbling up from the image parse in upstream's [`oci_util.go`](https://github.com/containers/image/blob/v5.35.0/oci/internal/oci_util.go#L32). If I had to guess, podman is providing their own upstack abstraction at that point. But something worth thinking about doing here so folks can avoid having to use `artifact`. Will mark it for roadmapping.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,7 @@
 ## UNRELEASED
 
 IMPROVEMENTS:
+* driver: Adds support for loading `oci-archive` and `docker-archive` images from `http(s)` URLs. [[GH-517](https://github.com/hashicorp/nomad-driver-podman/pull/517)]
 * config: Adds support for plugin configuration `networking.default_rootless_mode` to override the default networking mode when run rootlessly. [[GH-495](https://github.com/hashicorp/nomad-driver-podman/pull/495)]
 * config: Support custom Podman network names for static IP/MAC options via `network_mode`. [[GH-509](https://github.com/hashicorp/nomad-driver-podman/pull/509)]
 * config: Added support for `os`, `arch` and `variant` task config options to override the platform of the image to pull. [[GH-514](https://github.com/hashicorp/nomad-driver-podman/pull/514)]

--- a/README.md
+++ b/README.md
@@ -224,11 +224,18 @@ plugin "nomad-driver-podman" {
 
 ## Task Configuration
 
-* **image** - The image to run. Accepted transports are `docker` (default if missing), `oci-archive` and `docker-archive`. Images reference as [short-names](https://github.com/containers/image/blob/master/docs/containers-registries.conf.5.md#short-name-aliasing) will be treated according to user-configured preferences.
+* **image** - The image to run. Accepted transports are `docker` (default if missing), `oci-archive` and `docker-archive`. Images reference as [short-names](https://github.com/containers/image/blob/master/docs/containers-registries.conf.5.md#short-name-aliasing) will be treated according to user-configured preferences. The `oci-archive` and `docker-archive` transports also accept an `http(s)://` URL; the archive is downloaded (bounded by `image_pull_timeout`) and loaded into Podman.
 
 ```hcl
 config {
   image = "docker://redis"
+}
+```
+
+```hcl
+config {
+  # Load an image archive served over HTTP.
+  image = "oci-archive:https://example.com/images/redis.tar"
 }
 ```
 

--- a/api/image_inspect.go
+++ b/api/image_inspect.go
@@ -1,4 +1,4 @@
-// Copyright IBM Corp. 2019, 2025
+// Copyright IBM Corp. 2019, 2026
 // SPDX-License-Identifier: MPL-2.0
 
 package api

--- a/api/image_load.go
+++ b/api/image_load.go
@@ -1,4 +1,4 @@
-// Copyright IBM Corp. 2019, 2025
+// Copyright IBM Corp. 2019, 2026
 // SPDX-License-Identifier: MPL-2.0
 
 package api

--- a/api/image_load.go
+++ b/api/image_load.go
@@ -16,13 +16,21 @@ type imageLoadResponse struct {
 	Names []string `json:"Names"`
 }
 
-// ImageLoad uploads a tar archive as an image
+// ImageLoad uploads a tar archive on the local filesystem as an image.
 func (c *API) ImageLoad(ctx context.Context, path string) (string, error) {
 	archive, err := os.Open(path)
 	if err != nil {
 		return "", err
 	}
 	defer ignoreClose(archive)
+	return c.ImageLoadReader(ctx, archive)
+}
+
+// ImageLoadReader uploads a tar archive read from archive as an image. The
+// caller is responsible for closing archive. This allows loading an archive
+// from any source (e.g. a local file or an HTTP response body) without first
+// staging it on disk.
+func (c *API) ImageLoadReader(ctx context.Context, archive io.Reader) (string, error) {
 	response := imageLoadResponse{}
 
 	res, err := c.Post(ctx, "/v1.0.0/libpod/images/load", archive)

--- a/api/image_pull.go
+++ b/api/image_pull.go
@@ -1,4 +1,4 @@
-// Copyright IBM Corp. 2019, 2025
+// Copyright IBM Corp. 2019, 2026
 // SPDX-License-Identifier: MPL-2.0
 
 package api

--- a/driver.go
+++ b/driver.go
@@ -52,6 +52,11 @@ const (
 	// fingerprintPeriod is the interval at which the driver will send fingerprint responses
 	fingerprintPeriod = 30 * time.Second
 
+	// defaultHTTPTimeout bounds requests made by the driver's httpClient when
+	// downloading image archives from an http(s) URL. It matches the 60s used
+	// by the podman api client.
+	defaultHTTPTimeout = 60 * time.Second
+
 	// taskHandleVersion is the version of task handle which this driver sets
 	// and understands how to decode driver state
 	taskHandleVersion = 1
@@ -158,7 +163,7 @@ func NewPodmanDriver(logger hclog.Logger) drivers.DriverPlugin {
 		ctx:            ctx,
 		signalShutdown: cancel,
 		logger:         logger.Named(pluginName),
-		httpClient:     &http.Client{},
+		httpClient:     &http.Client{Timeout: defaultHTTPTimeout},
 	}
 }
 

--- a/driver.go
+++ b/driver.go
@@ -8,6 +8,7 @@ import (
 	"errors"
 	"fmt"
 	"net"
+	"net/http"
 	"os"
 	"path/filepath"
 	"runtime"
@@ -1263,9 +1264,20 @@ func (d *Driver) createImage(
 	var imageID string
 	imageName := image
 	loadedFromArchive := false
-	// If it is a shortname, we should not have to worry
-	// Let podman deal with it according to user configuration
-	if !shortnames.IsShortName(image) {
+	// Archive transports (oci-archive/docker-archive) pointing at an http(s)
+	// URL are handled here because the upstream containers/image reference
+	// parsers reject a URL in place of a local path. The archive is streamed
+	// straight to podman's load endpoint without staging it on disk.
+	if archiveURL, ok := archiveTransportURL(image); ok {
+		loadedName, err := d.loadImageFromURL(archiveURL, podmanClient, imagePullTimeout, cfg)
+		if err != nil {
+			return imageID, err
+		}
+		imageName = loadedName
+		loadedFromArchive = true
+	} else if !shortnames.IsShortName(image) {
+		// If it is a shortname, we should not have to worry
+		// Let podman deal with it according to user configuration
 		imageRef, err := parseImage(image)
 		if err != nil {
 			return imageID, fmt.Errorf("invalid image reference %s: %w", image, err)
@@ -1280,13 +1292,7 @@ func (d *Driver) createImage(
 			archiveData := imageRef.StringWithinTransport()
 			path := strings.Split(archiveData, ":")[0]
 			d.logger.Debug("Load image archive", "path", path)
-			_ = d.eventer.EmitEvent(&drivers.TaskEvent{
-				TaskID:    cfg.ID,
-				TaskName:  cfg.Name,
-				AllocID:   cfg.AllocID,
-				Timestamp: time.Now(),
-				Message:   "Loading image " + path,
-			})
+			d.emitImageEvent(cfg, "Loading image "+path)
 			imageName, err = podmanClient.ImageLoad(d.ctx, path)
 			if err != nil {
 				return imageID, fmt.Errorf("error while loading image: %w", err)
@@ -1315,13 +1321,7 @@ func (d *Driver) createImage(
 
 	d.logger.Info("Pulling image", "image", imageName)
 
-	_ = d.eventer.EmitEvent(&drivers.TaskEvent{
-		TaskID:    cfg.ID,
-		TaskName:  cfg.Name,
-		AllocID:   cfg.AllocID,
-		Timestamp: time.Now(),
-		Message:   "Pulling image " + imageName,
-	})
+	d.emitImageEvent(cfg, "Pulling image "+imageName)
 
 	pc := &registry.PullConfig{
 		Image:     imageName,
@@ -1367,6 +1367,75 @@ func (d *Driver) createImage(
 	imageID = result.(string)
 	d.logger.Debug("Pulled image ID", "imageID", imageID)
 	return imageID, nil
+}
+
+// emitImageEvent broadcasts a task event for an image operation (loading or
+// pulling), filling in the task identity fields shared by every such event.
+func (d *Driver) emitImageEvent(cfg *drivers.TaskConfig, message string) {
+	_ = d.eventer.EmitEvent(&drivers.TaskEvent{
+		TaskID:    cfg.ID,
+		TaskName:  cfg.Name,
+		AllocID:   cfg.AllocID,
+		Timestamp: time.Now(),
+		Message:   message,
+	})
+}
+
+// loadImageFromURL downloads an image archive from an http(s) URL and streams
+// it directly to podman's image load endpoint. The download is bounded by
+// imagePullTimeout and the archive is never staged on local disk. It returns
+// the name of the loaded image.
+func (d *Driver) loadImageFromURL(
+	url string,
+	podmanClient *api.API,
+	imagePullTimeout time.Duration,
+	cfg *drivers.TaskConfig,
+) (string, error) {
+	d.logger.Debug("Downloading image archive", "url", url)
+	d.emitImageEvent(cfg, "Loading image "+url)
+
+	ctx, cancel := context.WithTimeout(d.ctx, imagePullTimeout)
+	defer cancel()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return "", fmt.Errorf("failed to build request for image archive %s: %w", url, err)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("failed to download image archive %s: %w", url, err)
+	}
+
+	defer func() { _ = resp.Body.Close() }()
+
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("failed to download image archive %s: unexpected status %s", url, resp.Status)
+	}
+
+	imageName, err := podmanClient.ImageLoadReader(ctx, resp.Body)
+	if err != nil {
+		return "", fmt.Errorf("error while loading image: %w", err)
+	}
+	
+	return imageName, nil
+}
+
+// archiveTransportURL detects an oci-archive or docker-archive image reference
+// whose archive is located at an http(s) URL, e.g.
+// "oci-archive:http://host:9999/image.tar". The upstream containers/image
+// reference parsers cannot handle a URL in place of a local path, so these
+// references must be intercepted before parseImage. It returns the URL and true
+// when the reference matches, and "", false otherwise.
+func archiveTransportURL(image string) (string, bool) {
+	for _, transport := range []string{"oci-archive:", "docker-archive:"} {
+		if rest, ok := strings.CutPrefix(image, transport); ok {
+			if strings.HasPrefix(rest, "http://") || strings.HasPrefix(rest, "https://") {
+				return rest, true
+			}
+		}
+	}
+	return "", false
 }
 
 func parseImage(image string) (types.ImageReference, error) {

--- a/driver.go
+++ b/driver.go
@@ -1417,7 +1417,7 @@ func (d *Driver) loadImageFromURL(
 	if err != nil {
 		return "", fmt.Errorf("error while loading image: %w", err)
 	}
-	
+
 	return imageName, nil
 }
 

--- a/driver.go
+++ b/driver.go
@@ -128,6 +128,11 @@ type Driver struct {
 	// For any call where it's unspecified/unknown which podman should be used
 	defaultPodman *api.API
 
+	// httpClient is used to download image archives referenced by an http(s)
+	// URL. It is kept on the driver so request behaviour is configured in one
+	// place rather than relying on http.DefaultClient.
+	httpClient *http.Client
+
 	// singleflight group to prevent parallel image downloads
 	pullGroup singleflight.Group
 }
@@ -153,6 +158,7 @@ func NewPodmanDriver(logger hclog.Logger) drivers.DriverPlugin {
 		ctx:            ctx,
 		signalShutdown: cancel,
 		logger:         logger.Named(pluginName),
+		httpClient:     &http.Client{},
 	}
 }
 
@@ -1397,21 +1403,11 @@ func (d *Driver) loadImageFromURL(
 	ctx, cancel := context.WithTimeout(d.ctx, imagePullTimeout)
 	defer cancel()
 
-	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	resp, err := d.getImageArchive(ctx, url)
 	if err != nil {
-		return "", fmt.Errorf("failed to build request for image archive %s: %w", url, err)
+		return "", err
 	}
-
-	resp, err := http.DefaultClient.Do(req)
-	if err != nil {
-		return "", fmt.Errorf("failed to download image archive %s: %w", url, err)
-	}
-
 	defer func() { _ = resp.Body.Close() }()
-
-	if resp.StatusCode != http.StatusOK {
-		return "", fmt.Errorf("failed to download image archive %s: unexpected status %s", url, resp.Status)
-	}
 
 	imageName, err := podmanClient.ImageLoadReader(ctx, resp.Body)
 	if err != nil {
@@ -1419,6 +1415,30 @@ func (d *Driver) loadImageFromURL(
 	}
 
 	return imageName, nil
+}
+
+// getImageArchive issues the GET request for an image archive using the
+// driver's http client and returns the response on success. It encapsulates the
+// request boilerplate (building the request, performing it, and validating the
+// status) so the load logic stays separate from the transport concerns. The
+// caller is responsible for closing the response body.
+func (d *Driver) getImageArchive(ctx context.Context, url string) (*http.Response, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build request for image archive %s: %w", url, err)
+	}
+
+	resp, err := d.httpClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to download image archive %s: %w", url, err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		_ = resp.Body.Close()
+		return nil, fmt.Errorf("failed to download image archive %s: unexpected status %s", url, resp.Status)
+	}
+
+	return resp, nil
 }
 
 // archiveTransportURL detects an oci-archive or docker-archive image reference

--- a/driver_test.go
+++ b/driver_test.go
@@ -2440,6 +2440,51 @@ func createInspectImage(t *testing.T, image, reference string) {
 	must.Eq(t, idRef, idTest)
 }
 
+func Test_createImageArchivesHTTP(t *testing.T) {
+	ci.Parallel(t)
+
+	doesNotExist := func(filepath string) bool {
+		_, err := os.Stat(filepath)
+		if errors.Is(err, os.ErrNotExist) {
+			return true
+		}
+		must.NoError(t, err)
+		return false
+	}
+
+	if doesNotExist("/tmp/oci-archive") || doesNotExist("/tmp/docker-archive") {
+		t.Skip("Skipping image archive test. Missing prepared archive file(s).")
+	}
+
+	archiveDir := os.Getenv("ARCHIVE_DIR")
+	if archiveDir == "" {
+		t.Skip("Skipping image archive test. Missing \"ARCHIVE_DIR\" environment variable")
+	}
+
+	// Serve the prepared archive files over HTTP so createImage exercises the
+	// download-then-load path instead of reading from disk.
+	server := httptest.NewServer(http.FileServer(http.Dir(archiveDir)))
+	t.Cleanup(server.Close)
+
+	testCases := []struct {
+		Image     string
+		Reference string
+	}{
+		{
+			Image:     fmt.Sprintf("oci-archive:%s/oci-archive", server.URL),
+			Reference: "docker.io/library/alpine:latest",
+		},
+		{
+			Image:     fmt.Sprintf("docker-archive:%s/docker-archive", server.URL),
+			Reference: "docker.io/library/alpine:latest",
+		},
+	}
+
+	for _, testCase := range testCases {
+		createInspectImage(t, testCase.Image, testCase.Reference)
+	}
+}
+
 func createInspectImagePlatform(t *testing.T, image string, platform imagePlatform) string {
 	d := podmanDriverHarness(t, nil)
 
@@ -2697,6 +2742,83 @@ func Test_parseImage(t *testing.T) {
 		}
 
 	}
+}
+
+func Test_archiveTransportURL(t *testing.T) {
+	ci.Parallel(t)
+
+	testCases := []struct {
+		Name    string
+		Input   string
+		WantURL string
+		WantOK  bool
+	}{
+		{
+			Name:    "oci-archive http",
+			Input:   "oci-archive:http://10.211.55.9:9999/uploads/myhttp.tar",
+			WantURL: "http://10.211.55.9:9999/uploads/myhttp.tar",
+			WantOK:  true,
+		},
+		{
+			Name:    "oci-archive https",
+			Input:   "oci-archive:https://example.com/image.tar",
+			WantURL: "https://example.com/image.tar",
+			WantOK:  true,
+		},
+		{
+			Name:    "docker-archive http",
+			Input:   "docker-archive:http://example.com/image.tar",
+			WantURL: "http://example.com/image.tar",
+			WantOK:  true,
+		},
+		{
+			Name:   "docker-archive local path with tag",
+			Input:  "docker-archive:/tmp/image.tar:latest",
+			WantOK: false,
+		},
+		{
+			Name:   "docker transport url",
+			Input:  "docker://quay.io/repo/busybox:latest",
+			WantOK: false,
+		},
+		{
+			Name:   "plain image name",
+			Input:  "busybox:latest",
+			WantOK: false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.Name, func(t *testing.T) {
+			gotURL, gotOK := archiveTransportURL(testCase.Input)
+			must.Eq(t, testCase.WantOK, gotOK)
+			must.Eq(t, testCase.WantURL, gotURL)
+		})
+	}
+}
+
+// Test_loadImageFromURL_httpError verifies that a non-200 response from the
+// archive URL produces a clear error before any podman interaction.
+func Test_loadImageFromURL_httpError(t *testing.T) {
+	ci.Parallel(t)
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		http.Error(w, "no such archive", http.StatusNotFound)
+	}))
+	t.Cleanup(server.Close)
+
+	driver, ok := NewPodmanDriver(testlog.HCLogger(t)).(*Driver)
+	must.True(t, ok)
+
+	task := &drivers.TaskConfig{
+		ID:      uuid.Generate(),
+		Name:    "loadImageFromURL",
+		AllocID: uuid.Generate(),
+	}
+
+	_, err := driver.loadImageFromURL(server.URL+"/missing.tar", nil, time.Minute, task)
+	must.ErrorContains(t, err, "404")
+	must.ErrorContains(t, err, "unexpected status")
 }
 
 func Test_namedSocketIsDefault(t *testing.T) {

--- a/driver_test.go
+++ b/driver_test.go
@@ -2345,7 +2345,24 @@ func Test_createImage(t *testing.T) {
 	}
 
 	for _, testCase := range testCases {
-		createInspectImage(t, testCase.Image, testCase.Reference)
+		t.Run(testCase.Image, func(t *testing.T) {
+			d := podmanDriverHarness(t, nil)
+			driver := getPodmanDriver(t, d)
+
+			task := &drivers.TaskConfig{
+				ID:        uuid.Generate(),
+				Name:      "inspectImage",
+				AllocID:   uuid.Generate(),
+				Resources: createBasicResources(),
+			}
+
+			idTest, err := driver.createImage(testCase.Image, &TaskAuthConfig{}, false, false, imagePlatform{}, driver.defaultPodman, 5*time.Minute, task)
+			must.NoError(t, err)
+
+			idRef, err := driver.defaultPodman.ImageInspectID(context.Background(), testCase.Reference)
+			must.NoError(t, err)
+			must.Eq(t, idRef, idTest)
+		})
 	}
 }
 
@@ -2394,13 +2411,15 @@ func Test_createImageArchives(t *testing.T) {
 		return false
 	}
 
-	if doesNotExist("/tmp/oci-archive") || doesNotExist("/tmp/docker-archive") {
-		t.Skip("Skipping image archive test. Missing prepared archive file(s).")
-	}
-
+	// Archives are prepared by .github/machinesetup.sh in /tmp; ARCHIVE_DIR can
+	// override the location for local runs.
 	archiveDir := os.Getenv("ARCHIVE_DIR")
 	if archiveDir == "" {
-		t.Skip("Skipping image archive test. Missing \"ARCHIVE_DIR\" environment variable")
+		archiveDir = "/tmp"
+	}
+
+	if doesNotExist(archiveDir+"/oci-archive") || doesNotExist(archiveDir+"/docker-archive") {
+		t.Skip("Skipping image archive test. Missing prepared archive file(s).")
 	}
 
 	testCases := []struct {
@@ -2409,35 +2428,34 @@ func Test_createImageArchives(t *testing.T) {
 	}{
 		{
 			Image:     fmt.Sprintf("oci-archive:%s/oci-archive", archiveDir),
-			Reference: "docker.io/library/alpine:latest",
+			Reference: "docker.io/library/alpine:3",
 		},
 		{
 			Image:     fmt.Sprintf("docker-archive:%s/docker-archive", archiveDir),
-			Reference: "docker.io/library/alpine:latest",
+			Reference: "docker.io/library/alpine:3",
 		},
 	}
 
 	for _, testCase := range testCases {
-		createInspectImage(t, testCase.Image, testCase.Reference)
+		t.Run(testCase.Image, func(t *testing.T) {
+			d := podmanDriverHarness(t, nil)
+			driver := getPodmanDriver(t, d)
+
+			task := &drivers.TaskConfig{
+				ID:        uuid.Generate(),
+				Name:      "inspectImage",
+				AllocID:   uuid.Generate(),
+				Resources: createBasicResources(),
+			}
+
+			idTest, err := driver.createImage(testCase.Image, &TaskAuthConfig{}, false, false, imagePlatform{}, driver.defaultPodman, 5*time.Minute, task)
+			must.NoError(t, err)
+
+			idRef, err := driver.defaultPodman.ImageInspectID(context.Background(), testCase.Reference)
+			must.NoError(t, err)
+			must.Eq(t, idRef, idTest)
+		})
 	}
-}
-
-func createInspectImage(t *testing.T, image, reference string) {
-	d := podmanDriverHarness(t, nil)
-
-	task := &drivers.TaskConfig{
-		ID:        uuid.Generate(),
-		Name:      "inspectImage",
-		AllocID:   uuid.Generate(),
-		Resources: createBasicResources(),
-	}
-	driver := getPodmanDriver(t, d)
-	idTest, err := driver.createImage(image, &TaskAuthConfig{}, false, false, imagePlatform{}, driver.defaultPodman, 5*time.Minute, task)
-	must.NoError(t, err)
-
-	idRef, err := driver.defaultPodman.ImageInspectID(context.Background(), reference)
-	must.NoError(t, err)
-	must.Eq(t, idRef, idTest)
 }
 
 func Test_createImageArchivesHTTP(t *testing.T) {
@@ -2452,13 +2470,15 @@ func Test_createImageArchivesHTTP(t *testing.T) {
 		return false
 	}
 
-	if doesNotExist("/tmp/oci-archive") || doesNotExist("/tmp/docker-archive") {
-		t.Skip("Skipping image archive test. Missing prepared archive file(s).")
-	}
-
+	// Archives are prepared by .github/machinesetup.sh in /tmp; ARCHIVE_DIR can
+	// override the location for local runs.
 	archiveDir := os.Getenv("ARCHIVE_DIR")
 	if archiveDir == "" {
-		t.Skip("Skipping image archive test. Missing \"ARCHIVE_DIR\" environment variable")
+		archiveDir = "/tmp"
+	}
+
+	if doesNotExist(archiveDir+"/oci-archive") || doesNotExist(archiveDir+"/docker-archive") {
+		t.Skip("Skipping image archive test. Missing prepared archive file(s).")
 	}
 
 	// Serve the prepared archive files over HTTP so createImage exercises the
@@ -2472,16 +2492,33 @@ func Test_createImageArchivesHTTP(t *testing.T) {
 	}{
 		{
 			Image:     fmt.Sprintf("oci-archive:%s/oci-archive", server.URL),
-			Reference: "docker.io/library/alpine:latest",
+			Reference: "docker.io/library/alpine:3",
 		},
 		{
 			Image:     fmt.Sprintf("docker-archive:%s/docker-archive", server.URL),
-			Reference: "docker.io/library/alpine:latest",
+			Reference: "docker.io/library/alpine:3",
 		},
 	}
 
 	for _, testCase := range testCases {
-		createInspectImage(t, testCase.Image, testCase.Reference)
+		t.Run(testCase.Image, func(t *testing.T) {
+			d := podmanDriverHarness(t, nil)
+			driver := getPodmanDriver(t, d)
+
+			task := &drivers.TaskConfig{
+				ID:        uuid.Generate(),
+				Name:      "inspectImage",
+				AllocID:   uuid.Generate(),
+				Resources: createBasicResources(),
+			}
+
+			idTest, err := driver.createImage(testCase.Image, &TaskAuthConfig{}, false, false, imagePlatform{}, driver.defaultPodman, 5*time.Minute, task)
+			must.NoError(t, err)
+
+			idRef, err := driver.defaultPodman.ImageInspectID(context.Background(), testCase.Reference)
+			must.NoError(t, err)
+			must.Eq(t, idRef, idTest)
+		})
 	}
 }
 

--- a/registry/authentication.go
+++ b/registry/authentication.go
@@ -1,4 +1,4 @@
-// Copyright IBM Corp. 2019, 2025
+// Copyright IBM Corp. 2019, 2026
 // SPDX-License-Identifier: MPL-2.0
 
 package registry

